include/libcyphal/transport/udp/udp_transport_impl.hpp

```diff
@@ -692,22 +699,9 @@ class TransportImpl final : private TransportDelegate, public IUdpTransport  //
     {
         using PayloadFragment = cetl::span<const cetl::byte>;
 
-        while (const UdpardTxItem* const tx_item = ::udpardTxPeek(&media.udpard_tx()))
+        while (const UdpardTxItem* const tx_item = peekFirstValidUdpardTxItem(media.udpard_tx(), now))
         {
-            // We are dropping any TX item that has expired.
-            // Otherwise, we would send it to the media TX socket interface.
-            // We use strictly `>=` (instead of `>`) to give this frame a chance (one extra 1us) at the socket.
-            //
             const auto deadline = TimePoint{std::chrono::microseconds{tx_item->deadline_usec}};
-            if (now >= deadline)
-            {
-                // Release whole expired transfer b/c possible next frames of the same transfer are also expired.
-                popAndFreeUdpardTxItem(&media.udpard_tx(), tx_item, true /*whole transfer*/);
-
-                // No Sonar `cpp:S909` b/c it make sense to use `continue` statement here - the corner case of
-                // "early" (by deadline) transfer drop. Using `if` would make the code less readable and more nested.
-                continue;  // NOSONAR cpp:S909
```

Comment:
BTW, with this helper `peekFirstValidUdpardTxItem` method there is no need of NOSONAR anymore (for the `continue`).